api/fees/fees.go

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) 2025 The VeChainThor developers
+
+// Distributed under the GNU Lesser General Public License v3.0 software license, see the accompanying
+// file LICENSE or <https://www.gnu.org/licenses/lgpl-3.0.html>
+
+package fees
+
+import (
+	"math"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/vechain/thor/v2/api/utils"
+	"github.com/vechain/thor/v2/bft"
+	"github.com/vechain/thor/v2/chain"
+)
+
+type Fees struct {
+	data           *FeesData
+	bft            bft.Committer
+	backtraceLimit uint32 // The max number of blocks to backtrace.
+}
+
+func New(repo *chain.Repository, bft bft.Committer, backtraceLimit uint32, fixedCacheSize uint32) *Fees {
+	return &Fees{
+		data:           newFeesData(repo, fixedCacheSize),
+		bft:            bft,
+		backtraceLimit: backtraceLimit,
+	}
+}
+
+func (f *Fees) validateGetFeesHistoryParams(req *http.Request) (uint32, *chain.BlockSummary, error) {
+	// Validate blockCount
+	blockCountParam := req.URL.Query().Get("blockCount")
+	blockCount, err := strconv.ParseUint(blockCountParam, 10, 32)
+	if err != nil {
+		return 0, nil, utils.BadRequest(errors.WithMessage(err, "invalid blockCount, it should represent an integer"))
+	}
+
+	if blockCount == 0 {
+		return 0, nil, utils.BadRequest(errors.WithMessage(err, "invalid blockCount, it should not be 0"))
+	}
+
+	// Validate newestBlock
+	newestBlock, err := utils.ParseRevision(req.URL.Query().Get("newestBlock"), false)
+	if err != nil {
+		return 0, nil, utils.BadRequest(errors.WithMessage(err, "newestBlock"))
+	}
+
+	newestBlockSummary, err := utils.GetSummary(newestBlock, f.data.repo, f.bft)
+	if err != nil {
+		return 0, nil, utils.BadRequest(errors.WithMessage(err, "newestBlock"))
+	}
+
+	bestBlockSummary := f.data.repo.BestBlockSummary()
+	// Calculate minAllowedBlock
+	minAllowedBlock := uint32(math.Max(0, float64(int(bestBlockSummary.Header.Number())-int(f.backtraceLimit)+1)))
```

Comment:
Could you explain a bit regarding the +1 here? How do you define the range? Given best block to be 8000(back trace limit 1000), what's the valid range of allowed block does [7000, 800] sound valid to you?

Reply:
`minAllowedBlock` is used as a number of blocks (so it is kind of the "same family" as `blockCount`). That is the reason for + 1. You can check the usage below.

With `blockCount`, we will have 10 blocks for a range between 0 and 9. So `newestBlock` (`best`) would be 9 and blockCount >= 10 will return the 10 blocks.

Reply:
Just FYI all these validation have been simplified several times to the current state, and have been tested extensively, not only by the integration tests of this PR but also using the e2e test repo (PR in there to be created).

Feel free to suggest more integration tests if you think we should check an edge case 👍